### ISSUE TITLE
Username can contain invisible/directional characters

### ISSUE THREAD

**Issue by Kentaa** [infinity-next/infinity-next#65]
Usernames can contain invisible characters, such as a zero-width space, though this cannot be used (afaik) to register an account with a similar username (tested with MySQL.)

However, using the right-to-left invisible character can create a username which is visually identical to another.
The second line is actually vatsuG with a right-to-left character at the start.
```
Gustav
‮vatsuG
```

Other characters can be used which may not be wanted.
```
G͐̇̔u͇̮ͥ̒̔ͭ̾ͯͅṡ͔͍ͦ́̽̋ṱ͔̗̜̜̒͐̍̑͂a̙͔v̂ͣ̐ͫ̉͂
G͔̈̏̐̽ͅṳ̰͎̗́ͭ̽̔ͥș͔̫͎͈̣̅̓̆ͭ̓́͒t͖̖̭͍̳ͣͅͅa̦̖͉͇̹̳̳ͫ̍͒̆ͫͪ̒v̰̥̱̯̳͂̽̽͂̀͗ͯ
G͈̼͎̻̫̘u̗̣̜͒̌̉ś̺̾t̙̻͇̠͚͇͓ͥ̑ͮͯ̓̽͒ȁ̜̹v̪̗̯̤̦͉ͬ͑̽͒̐̏͒
```

**Comment by jaw-sh** [comment]
Pain in the ass, thanks.

I'm going to limit usernames to alphanumerics. That should solve all of these problems. Would you agree with that approach?

**Comment by Kentaa** [comment]
That would work, but perhaps later an option could be added to allow admins to set the character limit.

**Comment by jaw-sh** [comment]
My issue with any sort of blacklists is that you're going to end up creating loopholes. I'd rather just say, "your name is [at most] 64 characters and it will be a-z0-9." There's no way to screw that up and you always know what to expect in development.

**Comment by jaw-sh** [comment]
Fixed as a part of #62.
